doc/colvars-refman-main.tex

```diff
@@ -1009,42 +1012,58 @@
+\cvsubsubsec{Changing configuration upon restarting.}{}
 
-When restarting using binary state files, configuration changes are not allowed.  The easiest solution would be to produce a text-format state file specifically for that purpose.
-\ifdefined\cvscriptapi{%
-Alternatively, after restarting Colvars using a state file consistent with the previous configuration, the configuration may be changed using the scripting interface (see \ref{sec:cv_scripting}).
-}\fi
+In some cases, it is useful to modify the configuration of variables or biases between consecutive runs, for example by adding or removing a restraint.
+Some special provisions will happen in that case.
+When a state file is loaded, no information is available from in regarding the any newly added variable or bias, which will remain uninitialized until the first compute step.
+Conversely, any information that the state file contains about variables or biases that are no longer defined will be silently ignored.
+Because these checks are performed based only on the \emph{names} of variables and biases, it is your responsibility to ensure that these names have \emph{consistent definitions between runs.}
+
+The flexibility just described carries some limitations: namely, it is only supported when reading \emph{text-format} state files.
+Instead, restarting from binary files\cvlammpsonly{ (such as the LAMMPS restart file)}\cvgromacsonly{ (such as the GROMACS checkpoint file)} after a configuration change will trigger an error.
+It is also important to remind that when switching to a different build of \MDENGINE, the binary format may change slightly, even if the release version is the same.
+
+To work around the potential issues just described, a text-format Colvars state file should be used.
+\ifdefined\cvvmdornamd{This is the default in \MDENGINE{} unless the ``\texttt{COLVARS\_BINARY\_RESTART}'' is set to 1, and this information is only provided for troubleshooting.}\fi\cvlammpsonly{This can be achieved by providing an explicit \texttt{input} keyword when initializing the Colvars fix (see \ref{sec:colvars_mdengine_parameters}), which will instruct Colvars to use the given filename instead of the LAMMPS restart file.
+  Furthermore, the \texttt{fix\_modify} scripting command allows to load a Colvars file after initialization (\ref{sec:cv_command_loadsave}).}\cvgromacsonly{In GROMACS, this step would normally entail generating a new TPR file: however, there is no provision to embed a text-format Colvars state file in a TPR file.
+  For this specific use case, the following Colvars option may be used:
+\begin{itemize}
+\item %
+  \labelkey{Colvars-global|defaultInputStateFile}
+  \key{defaultInputStateFile}{%
+    global}{%
+    Default input state file, if not provided by \MDENGINE}{%
+    UNIX filename}{%
+    Define a state file that will be loaded by default, unless \MDENGINE{} provides restarting information through the checkpoint.
+  }
```

Comment:
This keyword is defined for all back-ends. Should it be part of the doc for all back-ends? Even if the other back-ends have other ways of specifying this, it could integrate better in some workflows.

Reply:
That's kind of related to my question to you about Tinker-HP. It's the only other engine where this keyword might be useful, I see really no use for it in LAMMPS, NAMD or VMD.

If it's really GROMACS specific right now, I would add a check to make it tied to it. We could always change it later if needed.